### ISSUE TITLE
Different output

### ISSUE THREAD

**Issue by WIll-Xu35** [Laulian/MxNet2Caffe-mobilefacenet#2]
Hi, I've used your tool to convert my trained resnet-50 model (using the official training code from insightface repository) and still got different output. The model was trained from scratch and its input was not 112x112, but 64x64.

After generating the prototxt, I manually modified the input shape to 1,3,64,64 and then generated the caffemodel.

The model can run without difficulty but the result is not even close to the mxnet result. I used the same image as output, also did preprocessing (image-127.5)/128. Changing the input image will result in a different output, so the network should be properly connected.

Any idea what might be the cause?

Many thanks.

**Comment by JCYAO-gz** [comment]
me too , mobilefacenet mxnet to caffe，same input，different output

**Comment by sky186** [comment]
@Laulian 
hi ,I want to convert arcface to caffe   ,now model last  is  Fully connected layer+bn(fc1)
fc1 = mx.sym.BatchNorm(data=key_i, fix_gamma=True, eps=2e-5, momentum=0.9, name='fc1')

now from data  to Fully connected layer  caffe out is right,
but the last bn is wrong,  (but I use author model all right ,I finutinue this save new model,not to modify model code,now only the last bn is wrong)  ,Could you help me ? I dont't know why?
the   code  as follow:
    ## last bn fc
    elif '_moving_mean' in key_i:
      key_caffe = key_i.replace('_moving_mean', '')
      net.params[key_caffe][0].data.flat = aux_params[key_i].asnumpy().flat
      net.params[key_caffe][2].data[...] = 1
    ## last bn fc
    elif '_moving_var' in key_i:
      key_caffe = key_i.replace('_moving_var', '')
      net.params[key_caffe][1].data.flat = aux_params[key_i].asnumpy().flat   #get weight
      net.params[key_caffe][2].data[...] = 1

##the caffe as follow
layer {
  bottom: "pre_fc1"
  top: "fc1"
  name: "fc1"
  type: "BatchNorm"
  batch_norm_param {
    use_global_stats: true
    moving_average_fraction: 0.9
    eps: 2e-05
  }
}
layer {
  bottom: "fc1"
  top: "fc1"
  name: "fc1_scale"
  type: "Scale"
  scale_param { bias_term: true }
}

##############
the test out,last bn out
 x: array([ 1.20850e+00,  3.41684e-01,  1.33299e+00,  1.69712e+00,
       -5.94079e-01,  5.64458e-01,  4.99271e-01, -4.05627e-01,
       -5.48919e-01, -8.93808e-02,  4.04169e-02,  7.23736e-01,...
 y: array([ 1.27053e+00,  3.60556e-01,  1.40213e+00,  1.78604e+00,
       -6.25261e-01,  5.95151e-01,  5.25208e-01, -4.26005e-01,
       -5.78033e-01, -9.44324e-02,  4.48017e-02,  7.61607e-01,...

**Comment by jeremycheong** [comment]
in the file mxnet2caffe.py, you should add the code
```python
    elif 'fc1_gamma' in key_i:
      net.params[key_caffe][0].data.flat = 1
```
in the all_keys loop.
if you want to transform the mobilefacenet, beacause in the last batchnormlization layer, the param fix_gamma is True.
